File: csharp/MapleStory.OpenAPI/Src/Dto/CharacterPropensityDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 캐릭터 성향 정보
    /// </summary>
    public class CharacterPropensityDTO
    {
        /// <summary>
        /// 조회 기준일 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
        /// </summary>
        [JsonProperty("date")]
        public string Date { get; set; }

        /// <summary>
        /// 카리스마 레벨
        /// </summary>
        [JsonProperty("charisma_level")]
        public long CharismaLevel { get; set; }

        /// <summary>
        /// 감성 레벨
        /// </summary>
        [JsonProperty("sensibility_level")]
        public long SensibilityLevel { get; set; }

        /// <summary>
        /// 통찰력 레벨
        /// </summary>
        [JsonProperty("insight_level")]
        public long InsightLevel { get; set; }

        /// <summary>
        /// 의지 레벨
        /// </summary>
        [JsonProperty("willingness_level")]
        public long WillingnessLevel { get; set; }

        /// <summary>
        /// 손재주 레벨
        /// </summary>
        [JsonProperty("handicraft_level")]
        public long HandicraftLevel { get; set; }

        /// <summary>
        /// 매력 레벨
        /// </summary>
        [JsonProperty("charm_level")]
        public long CharmLevel { get; set; }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/kms/dto/character/CharacterPropensityDTO.java
package dev.spiralmoon.maplestory.api.kms.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import dev.spiralmoon.maplestory.api.Utils;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.ZonedDateTime;

/**
 * 캐릭터 성향 정보
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterPropensityDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterPropensityDTO {

    /**
     * 조회 기준일 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
     */
    @JsonProperty("date")
    private String date;

    /**
     * 카리스마 레벨
     */
    @JsonProperty("charisma_level")
    private int charismaLevel;

    /**
     * 감성 레벨
     */
    @JsonProperty("sensibility_level")
    private int sensibilityLevel;

    /**
     * 통찰력 레벨
     */
    @JsonProperty("insight_level")
    private int insightLevel;

    /**
     * 의지 레벨
     */
    @JsonProperty("willingness_level")
    private int willingnessLevel;

    /**
     * 손재주 레벨
     */
    @JsonProperty("handicraft_level")
    private int handicraftLevel;

    /**
     * 매력 레벨
     */
    @JsonProperty("charm_level")
    private int charmLevel;

    /**
     * 조회 기준일 (KST, 일 단위 데이터로 시, 분은 일괄 0으로 표기)
     */
    public ZonedDateTime getDate() {
        return date != null
                ? Utils.toZonedDateTime(this.date)
                : null;
    }
}
